Add setting for more granular city feature code filtering

max-arnold opened the pull request:
I propose an explicit setting to filter cities based on their feature code (and not just 'PPL' substring). Rationale:

1. Substring matching by 'PPL' is hardcoded and is not flexible
2. Several feature codes (like PPLH) have no sense for majority of users (abandoned, historic or destroyed places)
3. PPLX is not even a city (city district or suburb)
4. STLMT is a city

Below is the frequency of each feature code in each database, along with its description:

feature code | cities1000 | cities5000 | cities15000 | description
-------------|------------|------------|-------------|-----------------------------------------------
 PPL   *     |  69474     |  29852     |  14395      | populated place, a city, town, village
 PPLA  *     |  3481      |  3481      |  2341       | seat of a first-order administrative division
 PPLA2 *     |  15796     |  6476      |  3457       | seat of a second-order administrative division
 PPLA3 *     |  27278     |  4996      |  2215       | seat of a third-order administrative division
 PPLA4 *     |  26591     |  900       |  151        | seat of a fourth-order administrative division
 PPLC  *     |  242       |  242       |  242        | capital of a political entity
 PPLCH       |  0         |  0         |  0          | historical capital of a political entity
 PPLF  *     |  6         |  2         |  1          | farm village
 PPLG  *     |  14        |  14        |  14         | seat of government of a political entity
 PPLH        |  6         |  2         |  2          | a populated place that no longer exists
 PPLL  *     |  239       |  82        |  33         | populated locality
 PPLQ        |  19        |  6         |  3          | abandoned populated place
 PPLR  *     |  4         |  2         |  1          | religious populated place
 PPLS  *     |  12        |  6         |  4          | populated places
 PPLW        |  1         |  1         |  1          | destroyed populated place
 PPLX        |  2338      |  1310      |  622        | section of populated place
 STLMT *     |  1         |  0         |  0          | israeli settlement
-------------------------------------------------------------------------------------------------------

This pull request alters amount of imported records due to different set of default feature codes (`CITIES_LIGHT_INCLUDE_CITY_TYPES` only contains the codes marked by asterisks in the table above).

jpic commented:
I think it makes sense to exclude non-cities by default, that would be considered a bugfix rather than a BC break.

PEP8 has some complains, see travis logs for details ;)

Overall this is a very good PR, thanks !!

max-arnold commented:
Fixed PEP8 issues, changed default INCLUDE_CITY_TYPES value to:
```
['PPL', 'PPLA', 'PPLA2', 'PPLA3', 'PPLA4', 'PPLC', 
'PPLF', 'PPLG', 'PPLL', 'PPLR', 'PPLS', 'STLMT']
```
If someone complains about different set of cities being imported, it is easy to restore the old behavior.

max-arnold commented:
Can I merge this one as well?

jpic commented:
Ok for merge B)

kodeine commented:
so i am trying to import the neighborhoods/hospitals as well, i am using `CITIES_LIGHT_INCLUDE_CITY_TYPES = ['PPLX', 'HSP', 'MKT', 'PPL', 'PPLA', 'PPLA2', 'PPLA3', 'PPLA4', 'PPLC', 'PPLF', 'PPLG', 'PPLL', 'PPLR', 'PPLS', 'STLMT']` but the town is not imported in db.
on geonames it's [this](http://www.geonames.org/search.html?q=model+town&country=PK)

it only imports 310 rows for cities. can anyone guide me to the right direction?

max-arnold commented:
@kodeine Please see issue #110. It mentions my experimental branch you can try.
